src/plugins/memdump/memdump.cpp

```diff
@@ -291,6 +293,122 @@ bool dump_memory_region(
     return ret;
 }
 
+static event_response_t terminate_process_hook_cb(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    // HANDLE ProcessHandle
+    uint64_t process_handle = drakvuf_get_function_argument(drakvuf, info, 1);
+
+    if (process_handle != 0xffffffffffffffffULL)
+    {
+        PRINT_DEBUG("[MEMDUMP] Process handle not pointing to self, ignore\n");
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    auto plugin = get_trap_plugin<memdump>(info);
+    if (!plugin)
+        return VMI_EVENT_RESPONSE_NONE;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+    access_context_t ctx =
+    {
+        .translate_mechanism = VMI_TM_PROCESS_DTB,
+        .dtb = info->regs->cr3
+    };
+
+    bool is_32bit;
+    addr_t stack_ptr;
+    addr_t frame_ptr;
+
+    if (drakvuf_get_user_stack32(drakvuf, info, &stack_ptr, &frame_ptr))
+    {
+        is_32bit = true;
+    }
+    else if (drakvuf_get_user_stack64(drakvuf, info, &stack_ptr))
+    {
+        is_32bit = false;
+    }
+    else
+    {
+        PRINT_DEBUG("[MEMDUMP] Failed to get stack pointer\n");
+        drakvuf_release_vmi(drakvuf);
+        return VMI_EVENT_RESPONSE_NONE;
+    }
+
+    PRINT_DEBUG("[MEMDUMP] Got stack pointer: %llx\n", (unsigned long long)stack_ptr);
+
+    size_t bytes_read = 0;
+    uint8_t buf[512];
+    ctx.addr = stack_ptr;
+    vmi_read(vmi, &ctx, 512, buf, &bytes_read);
+
+    for (size_t i = 0; i < bytes_read; i++)
+    {
+        addr_t stack_val = 0;
+        memcpy(&stack_val, buf+i, is_32bit ? 4 : 8);
+
+        mmvad_info_t mmvad;
+        if (drakvuf_find_mmvad(drakvuf, info->proc_data.base_addr, stack_val, &mmvad))
+        {
+            addr_t begin = mmvad.starting_vpn << 12;
+            size_t len = (mmvad.ending_vpn - mmvad.starting_vpn + 1) << 12;
+
+            page_info_t p_info = {0};
+
+            if (vmi_pagetable_lookup_extended(vmi, info->regs->cr3, stack_val, &p_info) == VMI_SUCCESS)
```

Comment:
Reverse the logic and add `continue` instead, reduces the indentation required for everything that follows

Reply:
done